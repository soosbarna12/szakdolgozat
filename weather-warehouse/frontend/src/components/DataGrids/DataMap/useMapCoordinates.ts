import { useEffect, useState } from "react";

export function useMapCoordinates(data: any) {
  const [coords, setCoords] = useState<[number, number]>([51.505, -0.09]);

  useEffect(() => {
    if (data && data.coord) {
      setCoords([data.coord.lat, data.coord.lon]);
    }
  }, [data]);

  return coords;
}
