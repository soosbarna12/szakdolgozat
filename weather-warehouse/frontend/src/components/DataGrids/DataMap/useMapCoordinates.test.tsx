import '@testing-library/jest-dom';
import { renderHook } from '@testing-library/react';
import { useMapCoordinates } from './useMapCoordinates';

describe('DataGrids/useMapCoordinates', () => {
    it('matches useMapCoordinates hook snapshot', () => {
        const { result } = renderHook(() => useMapCoordinates(null));
        expect(result.current).toMatchSnapshot();
    });

    it('returns default coordinates when no data is provided', () => {
        const { result } = renderHook(() => useMapCoordinates(null));
        expect(result.current).toEqual([51.505, -0.09]);
    });

    it('returns updated coordinates when valid data is provided', () => {
        const mockData = { coord: { lat: 40.7128, lon: -74.006 } };
        const { result } = renderHook(() => useMapCoordinates(mockData));
        expect(result.current).toEqual([40.7128, -74.006]);
    });

    it('updates coordinates when data changes', () => {
        const initialData = { coord: { lat: 40.7128, lon: -74.006 } };
        const updatedData = { coord: { lat: 34.0522, lon: -118.2437 } };

        const { result, rerender } = renderHook(({ data }) => useMapCoordinates(data), {
            initialProps: { data: initialData },
        });

        expect(result.current).toEqual([40.7128, -74.006]);

        rerender({ data: updatedData });
        expect(result.current).toEqual([34.0522, -118.2437]);
    });
});