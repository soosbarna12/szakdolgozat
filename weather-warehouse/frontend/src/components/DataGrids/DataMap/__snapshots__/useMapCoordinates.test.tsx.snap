// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataGrids/useMapCoordinates matches useMapCoordinates hook snapshot 1`] = `
[
  51.505,
  -0.09,
]
`;
